includes/ProductSync/ProductValidator.php

```diff
@@ -256,4 +289,79 @@ protected function validate_product_price() {
 		}
 	}
 
+	/**
+	 * Check if the description field has correct format according to:
+	 * Product Description Specifications for Catalogs : https://www.facebook.com/business/help/2302017289821154
+	 *
+	 * @throws ProductExcludedException If product description does not meet the requirements.
+	 */
+	protected function validate_product_description() {
+		/*
+		 * First step is to select the description that we want to evaluate.
+		 * Main description is the one provided for the product in the Facebook.
+		 * If it is blank, product description will be used.
+		 * If product description is blank, shortname will be used.
+		 */
+		$description = $this->facebook_product->get_fb_description();
+
+		/*
+		 * Requirements:
+		 * - No all caps descriptions.
+		 * - Max length 5000.
+		 * - Min length 30 ( tested and not required, will not enforce until this will become a hard requirement )
+		 */
+		if ( \WC_Facebookcommerce_Utils::is_all_caps( $description ) ) {
+			throw new ProductExcludedException( __( 'Product description is all capital letters. Please change the description to sentence case in order to allow synchronization of your product.', 'facebook-for-woocommerce' ) );
```

Comment:
I think it would be nice to introduce a different type of exception to handle these new validations because they are related to invalid data entry rather than to valid reasons like  "being out of stock" or "being part of an excluded category".

My suggestion would be something like this:
- `ProductInvalidException` for cases when the product fails validation for a reason that a property is invalid and "needs fixing by the user"  
- `ProductExcludedException` for cases when the product is excluded for an expected reason and doesn't need fixing 

This could be useful in the UI. For example if the product is "needs fixing" we could display a different color or icon to the user vs if the product is just excluded because it's out of stock.

Reply:
> This is a nice addition 👍
>
> This facebook-commerce.php file is 4000 lines long and is one of the messy dumping grounds in this plugin. It would be nice if we could extract this meta box into a separate class. Perhaps in the 'Admin' namespace?
I'm wondering if we can change the heading of the meta box to something more descriptive like "Facebook Product Feed"?


Agree. Done in 06737dc